fe/fe-core/src/main/java/org/apache/doris/job/ResourceQueue.java

```diff
@@ -0,0 +1,141 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.job;
+
+import org.apache.doris.common.UserException;
+import org.apache.doris.job.AsyncJob.JobStatus;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class ResourceQueue {
+    private static final AtomicLong QUEUE_ID = new AtomicLong(0);
+
+    private final long id;
+    private final String name;
+    private final ResourceQueueConfig config;
+    private final MatchingPolicy policy;
+    private final AsyncJobManager jobMgr;
```

Comment:
no need to call method of `jobMgr` in this class

Reply:
done